rust/src/linearview.rs

```diff
@@ -0,0 +1,515 @@
+        }
+    }
+
+    pub fn single_function_mlil(function: &Function, settings: &DisassemblySettings) -> Ref<Self> {
+        unsafe {
+            let handle = binaryninjacore_sys::BNCreateLinearViewSingleFunctionMediumLevelIL(
+                function.handle,
+                settings.handle,
+            );
+
+            Self::from_raw(handle)
+        }
+    }
+
+    pub fn single_function_mlil_ssa(
+        function: &Function,
+        settings: &DisassemblySettings,
+    ) -> Ref<Self> {
+        unsafe {
+            let handle = binaryninjacore_sys::BNCreateLinearViewSingleFunctionMediumLevelILSSAForm(
+                function.handle,
+                settings.handle,
+            );
+
+            Self::from_raw(handle)
+        }
+    }
+
+    pub fn single_function_hlil(function: &Function, settings: &DisassemblySettings) -> Ref<Self> {
+        unsafe {
+            let handle = binaryninjacore_sys::BNCreateLinearViewSingleFunctionHighLevelIL(
+                function.handle,
+                settings.handle,
+            );
+
+            Self::from_raw(handle)
+        }
+    }
+
+    pub fn single_function_hlil_ssa(
+        function: &Function,
+        settings: &DisassemblySettings,
+    ) -> Ref<Self> {
+        unsafe {
+            let handle = binaryninjacore_sys::BNCreateLinearViewSingleFunctionHighLevelILSSAForm(
+                function.handle,
+                settings.handle,
+            );
+
+            Self::from_raw(handle)
+        }
+    }
+
+    pub fn single_function_language_representation(
+        function: &Function,
+        settings: &DisassemblySettings,
+    ) -> Ref<Self> {
+        unsafe {
+            let handle =
+                binaryninjacore_sys::BNCreateLinearViewSingleFunctionLanguageRepresentation(
+                    function.handle,
+                    settings.handle,
+                );
+
+            Self::from_raw(handle)
+        }
+    }
+}
+
+unsafe impl RefCountable for LinearViewObject {
+    unsafe fn inc_ref(handle: &Self) -> Ref<Self> {
+        Ref::new(Self {
+            handle: BNNewLinearViewObjectReference(handle.handle),
+        })
+    }
+
+    unsafe fn dec_ref(handle: &Self) {
+        BNFreeLinearViewObject(handle.handle);
+    }
+}
+
+impl ToOwned for LinearViewObject {
+    type Owned = Ref<Self>;
+
+    fn to_owned(&self) -> Self::Owned {
+        unsafe { RefCountable::inc_ref(self) }
+    }
+}
+
+unsafe impl Send for LinearViewObject {}
+unsafe impl Sync for LinearViewObject {}
+
+pub struct LinearViewObjectIdentifier {
+    handle: *mut BNLinearViewObjectIdentifier,
+}
+
+impl LinearViewObjectIdentifier {
+    #[allow(unused)]
+    pub(crate) unsafe fn from_raw(handle: *mut BNLinearViewObjectIdentifier) -> Self {
+        debug_assert!(!handle.is_null());
+        Self { handle }
+    }
+}
+
+impl Drop for LinearViewObjectIdentifier {
+    fn drop(&mut self) {
+        unsafe { BNFreeLinearViewObjectIdentifier(self.handle) }
+    }
+}
```

Comment:
I don't think you use this, so you could remove it.  If you wanna have it thought, since it's just a struct:
```
	struct BNLinearViewObjectIdentifier
	{
		char* name;
		BNLinearViewObjectIdentifierType type;
		uint64_t start, end;
	};
```
It should be broken up/deserialized from C.  In this case the Python version is a good reference: `_to_core_struct` (to_raw?) and `_from_api_object` (from_raw?).

Reply:
Coming back to say that you already do this for `BNLinearDisassemblyLine`...so this should be similar.

Reply:
Removed in 23eb57d7874a9b7ffad4fb5d0c4f742eba845b50.